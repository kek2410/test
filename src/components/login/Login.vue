<template>
  <v-container style="max-width:350px;" name="loginCon">
    <v-row>
      <v-col cols="12">
        <v-alert :value="isLoginError" type="error">아이디와 비밀번호를 확인해주세요.</v-alert>
        <v-alert :value="isLogin" type="success">로그인이 완료되었습니다.</v-alert>
        <v-card>
          <v-toolbar flat>
            <v-toolbar-title>로그인</v-toolbar-title>
          </v-toolbar>
          <div class="pa-3">
            <v-text-field v-model="email" label="이메일을 입력해주세요." />
            <v-text-field
              v-model="password"
              type="password"
              label="패스워드를 입력하세요"
              keydown
              @keyup.13="loginGo"
            />
            <v-btn block depressed large @click="loginGo">로그인</v-btn>
          </div>
        </v-card>
      </v-col>
    </v-row>
  </v-container>
</template>

<script lang="ts">
import { Component, Vue } from "vue-property-decorator";
import { State, Getter, Action } from "vuex-class";
// import { State, Getter } from "vuex-class";
import { State as LoginState } from "@/store/modules/login/state";
// import { mapActions } from "vuex";

const namespace = "login";

@Component({
  computed: {
    // ...mapActions(namespace, ["actionLogin"])
  },
  methods: {
    // ...mapActions(namespace, ["actionLogin"])
  }
})
export default class LoginPage extends Vue {
  password: string = "";

  @State("login") login!: LoginState;
  @Action("changeState", { namespace }) changeState: any;

  get email() {
    console.log(this.login.email);
    return this.login.email;
  }

  set email(val: any) {
    // this.$store.state.login.email = val;
    this.changeState({ email: val });
  }

  get isLoginError() {
    return this.login.isLoginError;
  }

  get isLogin() {
    return this.login.isLogin;
  }

  @Action("actionLogin", { namespace }) actionLogin: any;
  @Getter("fullName", { namespace }) fullName!: string;

  mounted() {
    // fetching data as soon as the component's been mounted
  }

  loginGo(): void {
    const data = {
      email: this.email,
      password: this.password
    };
    this.actionLogin(data);
    // this.$store.dispatch("actionLogin", data, { root: true });
  }
}
</script>
