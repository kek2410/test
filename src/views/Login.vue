<template>
  <v-container style="max-width:350px;">
    <v-row>
      <v-col cols="12">
        <v-alert :value="isLoginError" type="error">
          아이디와 비밀번호를 확인해주세요.
        </v-alert>
        <v-alert :value="isLogin" type="success">
          로그인이 완료되었습니다.
        </v-alert>
        <v-card>
          <v-toolbar flat>
            <v-toolbar-title>로그인</v-toolbar-title>
          </v-toolbar>
          <div class="pa-3">
            <v-text-field v-model="email" label="이메일을 입력해주세요." />
            <v-text-field
              v-model="password"
              type="password"
              label="패스워드를 입력하세요"
            />
            <v-btn block depressed large @click="loginGo({ email, password })">
              로그인
            </v-btn>
          </div>
        </v-card>
      </v-col>
    </v-row>
  </v-container>
</template>

<script lang="ts">
import Vue from "vue";
import Component from "vue-class-component";
import { State, Getter, Action } from "vuex-class";
import { ProfileState } from "@/store/modules/login/state";

const namespace = "login";

@Component
export default class LoginPage extends Vue {
  email: string = "";
  password: string = "";

  @State("login") login!: ProfileState;

  get isLoginError() {
    return this.login.isLoginError;
  }

  get isLogin() {
    return this.login.isLogin;
  }

  @Action("login", { namespace }) actionLogin: any;
  @Getter("fullName", { namespace }) fullName!: string;

  mounted() {
    // fetching data as soon as the component's been mounted
  }

  loginGo(data: object): void {
    this.actionLogin(data);
  }
}
</script>
