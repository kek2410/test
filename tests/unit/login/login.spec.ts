// import { shallowMount, createLocalVue } from "@vue/test-utils";
import { mount, createLocalVue } from "@vue/test-utils";
import Login from "@/components/login/Login.vue";
import Vuex from "vuex";
// import { login as loginMudule } from "@/store/modules/login";
import Vuetify from "vuetify";

const localVue = createLocalVue();
localVue.use(Vuex);

describe("Login Store Module Test Go.", (): void => {
  let actions: any;
  let state: any;
  let store: any;
  let vuetify: any;

  beforeEach((): void => {
    vuetify = new Vuetify();

    state = {
      userInfo: {},
      isLogin: false,
      isLoginError: false
    };

    actions = {
      actionLogin: jest.fn(),
      actionLogOut: jest.fn()
    };

    store = new Vuex.Store({
      modules: {
        login: {
          state,
          actions
        }
      }
    });
  });

  it('calls store action "login" when button is clicked"', (): void => {
    const wrapper = mount(Login, { store, localVue, vuetify });
    const button = wrapper.find(".v-btn");

    // Here we bind a listener to the wrapper
    // instance to catch our custom event
    // https://vuejs.org/v2/api/#Instance-Methods-Events
    wrapper.vm.$on("action-btn:clicked", actions.actionLogin);

    expect(actions.actionLogin).toHaveBeenCalledTimes(0);

    // Simulate a click on the button
    button.trigger("click");

    // // Ensure that our mock event was called
    expect(actions.actionLogin).toHaveBeenCalledTimes(1);
  });

  // it('does not dispatch "actionInput" when event value is not "input"', (): void => {
  //   const wrapper = mount(Login, { store, localVue });
  //   const input = wrapper.find("input");
  //   // input.element.value = "not input";
  //   input.trigger("input");
  //   expect(actions.actionInput).not.toHaveBeenCalled();
  // });

  // it('calls store action "actionClick" when button is clicked', (): void => {
  //   const wrapper = mount(Login, { store, localVue });
  //   wrapper.find("button").trigger("click");
  //   expect(actions.actionClick).toHaveBeenCalled();
  // });
});
