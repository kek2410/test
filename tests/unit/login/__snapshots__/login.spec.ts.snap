// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Store Module Test Go. Login component - Snapshot 1`] = `
<div
  class="container name"
  style="max-width: 350px;"
>
  <div
    class="row"
  >
    <div
      class="col col-12"
    >
      <div
        class="v-alert v-sheet theme--dark error"
        role="alert"
        style="display: none;"
      >
        <div
          class="v-alert__wrapper"
        >
          <i
            aria-hidden="true"
            class="v-icon notranslate v-alert__icon mdi mdi-alert theme--dark"
          />
          <div
            class="v-alert__content"
          >
            아이디와 비밀번호를 확인해주세요.
          </div>
        </div>
      </div>
       
      <div
        class="v-alert v-sheet theme--dark success"
        role="alert"
        style="display: none;"
      >
        <div
          class="v-alert__wrapper"
        >
          <i
            aria-hidden="true"
            class="v-icon notranslate v-alert__icon mdi mdi-check-circle theme--dark"
          />
          <div
            class="v-alert__content"
          >
            로그인이 완료되었습니다.
          </div>
        </div>
      </div>
       
      <div
        class="v-card v-sheet theme--light"
      >
        <header
          class="v-sheet v-sheet--tile theme--light v-toolbar v-toolbar--flat"
          style="height: 64px;"
        >
          <div
            class="v-toolbar__content"
            style="height: 64px;"
          >
            <div
              class="v-toolbar__title"
            >
              로그인
            </div>
          </div>
        </header>
         
        <div
          class="pa-3"
        >
          <div
            class="v-input theme--light v-text-field"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-text-field__slot"
                >
                  <label
                    class="v-label theme--light"
                    for="input-11"
                    style="left: 0px; position: absolute;"
                  >
                    이메일을 입력해주세요.
                  </label>
                  <input
                    id="input-11"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <span
                    class="v-messages__wrapper"
                    name="message-transition"
                    tag="div"
                  />
                </div>
              </div>
            </div>
          </div>
           
          <div
            class="v-input theme--light v-text-field"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-text-field__slot"
                >
                  <label
                    class="v-label theme--light"
                    for="input-14"
                    style="left: 0px; position: absolute;"
                  >
                    패스워드를 입력하세요
                  </label>
                  <input
                    id="input-14"
                    keydown=""
                    type="password"
                  />
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <span
                    class="v-messages__wrapper"
                    name="message-transition"
                    tag="div"
                  />
                </div>
              </div>
            </div>
          </div>
           
          <button
            class="v-btn v-btn--block v-btn--depressed theme--light v-size--large"
            type="button"
          >
            <span
              class="v-btn__content"
            >
              로그인
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
